From: Latchesar Ionkov <lucho@ionkov.net>
Subject: [PATCH 042/111]     9p: Pass the correct end of buffer to p9stat_read
Git-commit: da7ddd3296505b4cb46685e1bbf7d0075b3cd4f1
Patch-Mainline: 2.6.35
References: FATE#311639

    Pass the correct end of the buffer to p9stat_read.

	upstream commit: da7ddd3296505b4cb46685e1bbf7d0075b3cd4f1

    Signed-off-by: Latchesar Ionkov <lucho@ionkov.net>
    Signed-off-by: Eric Van Hensbergen <ericvh@gmail.com>
Signed-off-by: Harsh Prateek Bora <harsh@linux.vnet.ibm.com>
Acked-by: Bruce Rogers <brogers@suse.com>
---
 fs/9p/vfs_dir.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/fs/9p/vfs_dir.c b/fs/9p/vfs_dir.c
index 13ce197..be9a756 100644
--- a/fs/9p/vfs_dir.c
+++ b/fs/9p/vfs_dir.c
@@ -145,7 +145,7 @@ static int v9fs_dir_readdir(struct file *filp, void *dirent, filldir_t filldir)
 		while (rdir->head < rdir->tail) {
 			p9stat_init(&st);
 			err = p9stat_read(rdir->buf + rdir->head,
-						buflen - rdir->head, &st,
+						rdir->tail - rdir->head, &st,
 						fid->clnt->proto_version);
 			if (err) {
 				P9_DPRINTK(P9_DEBUG_VFS, "returned %d\n", err);
-- 
1.7.1.1

